compute_endpoint/globus_compute_endpoint/engines/high_throughput/interchange.py

```diff
@@ -977,23 +977,12 @@ def start(self, poll_period: int | None = None) -> None:
 
                                 mdata["tasks"][task_container].remove(tid)
 
-                                # Transfer any outstanding task statuses to the
-                                # result message
                                 if tid in self.task_status_deltas:
-                                    r["task_statuses"] += self.task_status_deltas[tid]
                                     del self.task_status_deltas[tid]
-                                    b_messages[idx] = dill.dumps(r)
```

Comment:
Och this is a nice cleanup!  @ryanchard : maybe verify this still works per your expectations?

Reply:
If I understand correctly, this goes back to throwing away all task status deltas, rather than moving them into the response message? If the delta was just the `running` status, that would be ok as it isn't useful after completion anyway. However, don't we also have `waiting-for-node` and `waiting-for-launch` in here? We shouldn't throw them away if they are there.

Reply:
Fixed. I've removed the `del` statement so that a task's deltas would still propagate even if it might be after the task completion is registered by the hosted services.

Reply:
For posterity, we've had a side-band conversation about this.  At the point we're not removing the entry from the `self.task_status_deltas` data structure, we might as well go all the way and let the TSD data structure handle the task status parts it has.  No need to coalesce in either direction.  Thus, this if-block can be entirely removed, and then the need for the lock-block is also removed.